### ISSUE TITLE
[Feedback] Split Icons into little ESMs

### ISSUE THREAD

**Issue by daaku** [primer/octicons#610]
It would be ideal if each icon were a standalone ESM file. The @primer/components library seems to have taken this approach and it makes tree shaking a deterministic process, and works with stricter dead code elimination (for example currently due to `defaultProps` every icon may in theory have side effects).

**Comment by colebemis** [comment]
This library should be tree-shaking compatible. Is it not working for you?

**Comment by daaku** [comment]
Indeed, I was trying to use it with esbuild, which has stricter tree shaking than rollup/wepack. More context here: https://github.com/evanw/esbuild/issues/1241
